avionics/common/include/rocket.h

```diff
@@ -0,0 +1,124 @@
+        switch (command) {
+        case 'P':
+            radio.sendStatus(new_time_int, init_status, sensors, ignitors);
+            break;
+        case 'A':
+            state_machine.arm();
+            break;
+        case 'C':
+            radio.sendConfig(new_time_int);
+            break;
+        case 'D':
+            state_machine.disarm();
+            break;
+        case 0x30:
+            radio.sendBulkSensor(new_time_int, calc.altitude(),
+                                 sensors.accelerometer, sensors.imuSensor,
+                                 sensors.gps, static_cast<uint8_t>(state));
+            break;
+        case 0x04:
+            radio.sendGPS(new_time_int, sensors.gps);
+            break;
+        case 0x10:
+            radio.sendSingleSensor(new_time_int, 0x10,
+                                   sensors.accelerometer.getData()[0]);
+            break;
+
+        case 0x11:
+            radio.sendSingleSensor(new_time_int, 0x11,
+                                   sensors.accelerometer.getData()[1]);
+            break;
+        case 0x12:
+            radio.sendSingleSensor(new_time_int, 0x12,
+                                   sensors.accelerometer.getData()[2]);
+            break;
+        case 0x13:
+            radio.sendSingleSensor(new_time_int, 0x13,
+                                   sensors.barometer.getData()[0]);
+            break;
+        case 0x14:
+            radio.sendSingleSensor(new_time_int, 0x14,
+                                   sensors.barometer.getData()[1]);
+            break;
+        case 0x15:
+            radio.sendSingleSensor(new_time_int, 0x15,
+                                   sensors.temperature.getData()[0]);
+            break;
+        case 0x16:
+            break;
+            // Not implemented - still uncertain about IMU storage
+        case 0x17:
+            break;
+            // Not implemented - still uncertain about IMU storage
+        case 0x18:
+            break;
+            // Not implemented - still uncertain about IMU storage
+        case 0x19:
+            radio.sendSingleSensor(new_time_int, 0x19,
+                                   sensors.gps.getData()[0]);
+            break;
+        case 0x1A:
+            radio.sendSingleSensor(new_time_int, 0x1A,
+                                   sensors.gps.getData()[1]);
+            break;
+        case 0x1B:
+            radio.sendSingleSensor(new_time_int, 0x1B,
+                                   sensors.gps.getData()[2]);
+            break;
+        case 0x1C:
+            radio.sendSingleSensor(new_time_int, 0x1C, calc.altitude());
+            break;
+        case 0x1D:
+            radio.sendState(new_time_int, static_cast<uint8_t>(state));
+            break;
+        case 0x1E:
+            break; // Voltage sensor not implemented
+        case 0x1F:
+            radio.sendSingleSensor(new_time_int, 0x1F, calc.getBaseAltitude());
+            break;
+        default:
+            break;
```

Comment:
Should probably throw an exception or otherwise alert the user that they've entered an invalid command.

Reply:
Throwing an exception would terminate the program on what is ultimately a non-fatal error - although I'd be open to logging it in a file somewhere.

Reply:
I'm not saying we 100% should do this, but one idea would be to catch all exceptions in the main loop and send out some and event, message, or other radio packet.

Would be nice to make things fail more loudly. Might help us catch more bugs in CI.

Or we could let execution terminate and fail that way. That way not only would it fail loudly but we could produce a core dump to help with debugging. What I've seen done (usually with asserts) is they are only enabled simulation/testing builds. That way as you say, we don't crash on non-fatal errors when it matters, and improve performance, but we retain the ability to fail loudly in testing.

Second option might be cleaner and more useful. If we go with any at all.

What do you guys think?

Reply:
Exceptions are particularly useful if the function/scope you're in doesn't have enough info to recover from the problem at hand - but in this case, we do actually have all the info needed to deal with the problem, since we have access to the `rocket` object. (Although it'd be reasonable to pull this into a different error-handling function, to keep the command receiving function focused)

 So I think it'd still be better to log the event here and now, rather than skip over the rest of the code. One option is to send a SIM error packet on failure, so the ground station can terminate gracefully.

Reply:
In any case though - in the interest of keeping pull requests focused, it might be better to have this be another ticket, rather than gluing it on here.

Reply:
https://trello.com/c/nFfK9weQ/294-figure-out-what-can-be-done-about-xbee-errors

Reply:
I like the idea of a SIM error packet so that we're not ignoring errors.